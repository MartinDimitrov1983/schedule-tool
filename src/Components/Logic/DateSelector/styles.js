const styles = {
  period: {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr 1fr 56.75%',
    gap: '10px',
  },
  daysLength: { display: 'flex', alignItems: 'end', marginBottom: '11px' },
  buttonWrapper: {
    display: 'flex',
    gap: '16px',
    justifyContent: 'end',
    alignItems: 'end',
  },
  button: (disabled) => {
    return {
      width: '50px',
      height: '50px',
      margin: '-10px',
      cursor: `${!disabled && 'pointer'}`,
      fill: `${disabled ? '#acacac' : 'black'}`,
    };
  },
};

export default styles;
