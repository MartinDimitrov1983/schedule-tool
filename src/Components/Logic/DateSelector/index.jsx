import React from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { Box, Typography } from '@mui/material';
import { DatePicker } from '../..';
import { Back, Forward } from '../../../Assets';
import {
  changeStartDate,
  changeEndDate,
  changeDays,
} from '../../../Redux/slices/days';
import { formatDate, getDatesAndDays } from '../../../Utils';
import { DAYS, END_DATE, START_DATE, WEEK_LENGTH } from '../../../Constants';
//calculateEndDate
import styles from './styles';

const DateSelector = ({
  backHandler,
  forwardHandler,
  startDateCount,
  endDateCount,
}) => {
  const startDate = useSelector((state) => state.days.startDate);
  const endDate = useSelector((state) => state.days.endDate);
  const days = useSelector((state) => state.days.days);

  const dispatch = useDispatch();

  const startDateHandler = (value) => {
    const formatedDate = formatDate(value);

    dispatch(changeStartDate(formatedDate));
    dispatch(changeDays(getDatesAndDays(formatedDate, endDate)));
  };

  const endDateHandler = (value) => {
    const formatedDate = formatDate(value);
    dispatch(changeEndDate(formatedDate));
    dispatch(changeDays(getDatesAndDays(startDate, formatedDate)));
  };

  return (
    <Box sx={styles.period}>
      <DatePicker
        title={START_DATE}
        onChange={startDateHandler}
        date={startDate ? new Date(startDate) : null}
      />
      <DatePicker
        title={END_DATE}
        onChange={endDateHandler}
        date={endDate ? new Date(endDate) : null}
      />
      <Box sx={styles.daysLength}>
        {days.length > 0 && <Typography>{`${days.length} ${DAYS}`}</Typography>}
      </Box>
      <Box sx={styles.buttonWrapper}>
        {days.length > WEEK_LENGTH && (
          <>
            <Back
              style={styles.button(startDateCount === 0)}
              onClick={backHandler}
              disabled={startDateCount === 0}
            />
            <Forward
              style={styles.button(endDateCount === days.length)}
              onClick={forwardHandler}
              disabled={endDateCount === days.length}
            />
          </>
        )}{' '}
      </Box>
    </Box>
  );
};

DateSelector.propTypes = {
  backHandler: PropTypes.func,
  forwardHandler: PropTypes.func,
  startDateCount: PropTypes.number,
  endDateCount: PropTypes.number,
};

export default DateSelector;
