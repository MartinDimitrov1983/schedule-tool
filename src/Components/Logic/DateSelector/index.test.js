import React from 'react';
import { render } from '@testing-library/react';
import * as Redux from 'react-redux';

import DateSelector from './index';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

test('Snapshot DateSelector', () => {
  jest.spyOn(Redux, 'useSelector').mockReturnValue([
    { date: '2024-04-17', day: 'Wednesday' },
    { date: '2024-04-18', day: 'Thursday' },
    { date: '2024-04-19', day: 'Friday' },
  ]);
  jest.spyOn(Redux, 'useSelector').mockReturnValue('2024-04-17');

  const { asFragment } = render(
    <DateSelector
      backHandler={() => {}}
      forwardHandler={() => {}}
      startDateCount={2}
      endDateCount={3}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
