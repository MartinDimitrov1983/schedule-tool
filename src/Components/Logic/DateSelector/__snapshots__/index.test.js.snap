// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot DateSelector 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-19iv8nx"
  >
    <div
      class="MuiBox-root css-dvxtzn"
    >
      <p>
        Start-Date
      </p>
      <div
        class="react-date-picker react-date-picker--closed react-date-picker--enabled datePicker"
      >
        <div
          class="react-date-picker__wrapper"
        >
          <div
            class="react-date-picker__inputGroup"
          >
            <input
              hidden=""
              max="275760-09-13"
              min="0001-01-01"
              name="date"
              style="visibility: hidden; position: absolute; z-index: -999;"
              type="date"
              value="2024-04-17"
            />
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputmode="numeric"
              max="30"
              min="1"
              name="day"
              type="number"
              value="17"
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              .
            </span>
            <span
              class="react-date-picker__inputGroup__leadingZero"
            >
              0
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month react-date-picker__inputGroup__input--hasLeadingZero"
              data-input="true"
              inputmode="numeric"
              max="12"
              min="1"
              name="month"
              type="number"
              value="4"
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              .
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputmode="numeric"
              max="275760"
              min="1"
              name="year"
              step="1"
              type="number"
              value="2024"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-dvxtzn"
    >
      <p>
        End-Date
      </p>
      <div
        class="react-date-picker react-date-picker--closed react-date-picker--enabled datePicker"
      >
        <div
          class="react-date-picker__wrapper"
        >
          <div
            class="react-date-picker__inputGroup"
          >
            <input
              hidden=""
              max="275760-09-13"
              min="0001-01-01"
              name="date"
              style="visibility: hidden; position: absolute; z-index: -999;"
              type="date"
              value="2024-04-17"
            />
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputmode="numeric"
              max="30"
              min="1"
              name="day"
              type="number"
              value="17"
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              .
            </span>
            <span
              class="react-date-picker__inputGroup__leadingZero"
            >
              0
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month react-date-picker__inputGroup__input--hasLeadingZero"
              data-input="true"
              inputmode="numeric"
              max="12"
              min="1"
              name="month"
              type="number"
              value="4"
            />
            <span
              class="react-date-picker__inputGroup__divider"
            >
              .
            </span>
            <input
              autocomplete="off"
              class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputmode="numeric"
              max="275760"
              min="1"
              name="year"
              step="1"
              type="number"
              value="2024"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-t3kwy6"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        10 days
      </p>
    </div>
    <div
      class="MuiBox-root css-drtte9"
    >
      <svg
        style="width: 50px; height: 50px; margin: -10px; cursor: pointer; fill: black;"
      >
        back.svg
      </svg>
      <svg
        style="width: 50px; height: 50px; margin: -10px; cursor: pointer; fill: black;"
      >
        forward.svg
      </svg>
       
    </div>
  </div>
</DocumentFragment>
`;
